src/new_index/mempool.rs

```diff
@@ -285,6 +285,48 @@ impl Mempool {
         &self.backlog_stats.0
     }
 
+    pub fn old_txids(&self) -> HashSet<Txid> {
+        return HashSet::from_iter(self.txstore.keys().cloned());
+    }
+
+    pub fn download_new_mempool_txs(daemon: &Daemon, old_txids: &HashSet<Txid>, new_txids: &HashSet<Txid>) -> Vec<Transaction> {
+        let t = Instant::now();
+
+        let txids: Vec<&Txid> = (*new_txids).difference(old_txids).collect();
+        let tranactions = match daemon.gettransactions(&txids) {
+            Ok(txs) => txs,
+            Err(err) => {
+                warn!("failed to get {} transactions: {}", txids.len(), err); // e.g. new block or RBF
+                vec![] // return an empty vector if there's an error
```

Comment:
Previously, failing to fetch all mempool transactions resulted in the entire `update()` operation getting skipped until the next scheduled run. Was changing this intentional?

The way its implemented now means that we could evict transactions from the mempool without adding the new ones that replaced them, which would make our view of the mempool inconsistent with the bitcoind mempool. Before this change, our view would remain outdated when failing to fetch transactions, but still be consistent with the latest bitcoind mempool we were able to sync with.

The new behavior may actually be better, but this requires some thought. Also, this is important enough of a change that if we do change it, I would mention it in the commit message.

Reply:
Good catch -- this change was actually unintentional. Thanks!

Reply:
> The new behavior may actually be better

To expand a little bit on this, it seems that only removing transactions known not to be in the mempool anymore can be preferable over the current behavior, especially following a new block - we'd want the transactions included in the block removed from our view of the mempool asap, to avoid situations where a transaction exists in both places.

This will result in inconsistency with bitcoind as I mentioned in the previous comment, but it seems to be a safe inconsistency.

If this was the other way around, where we might admit new transactions into the mempool without removing the evicted ones they replaced, we could end up in a situation where our view of the mempool has two conflicting double-spend transactions, which would be unsafe.

But in our case, where we might remove evicted transactions without admitting the transactions that replaced them, it seems that nothing bad can happen?